docs/process/release-planning.md

```diff
@@ -1,7 +1,71 @@
 # Release Planning
 
-We plan work at two levels of granularity: first at the marketing release level (1.1, 1.2, etc.) and then at the milestone level. Marketing releases are made up of multiple milestones. Milestones should contain roughly a month’s worth of work.
+This document outlines our process for planning and scheduling releases, so you
+can familiarize yourself with the flow of work from opening an issue to seeing
+it published in a release.
 
-We’ll release updates as needed or when a milestone is completed.
+## Releases
 
-Large features and user-visible changes are feature-flagged, to be unflagged in the next marketing release. Beta and dev builds will have all feature flags enabled.
+We organize releases in two ways - marketing and milestones:
+
+ - Marketing releases are what we use to represent planned features, and are
+   high-level goals
+      - **for example: 1.4, 1.5, etc.**
+ - Milestone releases are used to track issues and pull requests associated
+   with a marketing release, and can be followed along on GitHub
+      - **for example: 1.4.1, 1.4.2, 1.5.0, etc.**
+
+We aim to ship updates to production approximately every two weeks, to ensure a continuous
+flow of improvements to our users. Track our progress in the [current milestones](https://github.com/desktop/desktop/milestones?direction=desc&sort=completeness&state=open).
+
+## Scheduling Pull Requests
+
+Pull Requests for user-facing changes should have a milestone associated with
+them, to indicate when they should be merged.
+
+### Features
+
+Pull Requests associated with features for our marketing releases should have a
+milestone defined as soon as possible, to indicate the anticipated release and
+help track.
+
+These pull requests should also be behind a feature flag, so we can control when
+a feature is enabled for users. If you are using the GitHub Desktop
+[beta channel](https://github.com/desktop/desktop#beta-channel)
+you will be able to help test and provide feedback about these features early.
+
+### Bugfixes
+
+Pull Requests associated with bugfixes or unplanned work can be opened early,
+but **should not** be assigned a milestone until after they have been reviewed and
+approved.
```

Comment:
this is after they've been reviews/approved and then the 24 hour wait window begins, correct? 

i realized we aren't explicit about the wait window in this doc and it might be helpful to call it out. this implies that other contributors/maintainers have 24h to discuss whether the milestone could be changed

Reply:
> this is after they've been reviews/approved and then the 24 hour wait window begins, correct?

Not a strong feel on this, but I'm fine with someone proposing the milestone after approving (for example "I'd like this to go out with the next bugfix release"), assigning the milestone (for visibility) and incorporating that into our existing 24 hour wait window as a discussion point.

> this implies that other contributors/maintainers have 24h to discuss whether the milestone could be changed

Yep, and we can make that explicit if we're on the same page.

Reply:
See cce8e9d874b059f537edb4856ab7739b76596c8f